integrationtest/mapper/orm-jakarta-batch/src/test/java/org/hibernate/search/integrationtest/jakarta/batch/massindexing/MassIndexingJobWithCompositeIdIT.java

```diff
@@ -54,30 +52,25 @@
  *
  * @author Mincong Huang
  */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
 @TestForIssue(jiraKey = "HSEARCH-2615")
-public class MassIndexingJobWithCompositeIdIT {
+class MassIndexingJobWithCompositeIdIT {
 
 	private static final LocalDate START = LocalDate.of( 2017, 6, 1 );
 
 	private static final LocalDate END = LocalDate.of( 2017, 8, 1 );
 
-	@ClassRule
-	public static ReusableOrmSetupHolder setupHolder =
-			ReusableOrmSetupHolder.withSingleBackend( BackendConfigurations.simple() );
-	@Rule
-	public MethodRule setupHolderMethodRule = setupHolder.methodRule();
+	@RegisterExtension
+	public static OrmSetupHelper ormSetupHelper =
+			OrmSetupHelper.withSingleBackend( BackendConfigurations.simple() );
 
 	private EntityManagerFactory emf;
 
-	@ReusableOrmSetupHolder.Setup
-	public void setup(OrmSetupHelper.SetupContext setupContext) {
-		setupContext.withAnnotatedTypes( EntityWithIdClass.class, EntityWithEmbeddedId.class )
-				.withProperty( HibernateOrmMapperSettings.INDEXING_LISTENERS_ENABLED, false );
-	}
-
-	@Before
-	public void initData() throws Exception {
-		emf = setupHolder.entityManagerFactory();
+	@BeforeAll
+	void setup() {
```

Comment:
I read something about non-static `@BeforeAll` methods only working with JDK 16+... you might want to try running this on JDK 11 before you do this everywhere.

Reply:
Hmm, I've tested this by running tests with jdk11 locally and it seems fine...
I went with this approach mostly to get build failures if a `@TestInstance(PER_CLASS)`  is missing on a test with `@BeforeAll` as we'd almost always need to have it if we do the before-all initialization. But let's run the jdk11 build on CI once we are done with everything else, and if there are any problems I'll update it to a static method.